forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,326 @@
+use time::{format_description::well_known::Iso8601, Date, OffsetDateTime};
+
+pub(crate) const OUTPUT_PATH_DEFAULT_FORMAT: &str =
+    "forest_snapshot_{chain}_{year}-{month}-{day}_height_{height}.car";
+
+#[derive(Debug, StructOpt)]
+pub enum SnapshotCommands {
+    /// Export a snapshot of the chain to `<output_path>`
+    Export {
+        /// Tipset to start the export from, default is the chain head
+        #[structopt(short, long)]
+        tipset: Option<i64>,
+        /// Specify the number of recent state roots to include in the export.
+        #[structopt(short, long, default_value = "2000")]
+        recent_stateroots: i64,
+        /// Include old messages
+        #[structopt(short, long)]
+        include_old_messages: bool,
+        /// Snapshot output path. Default to `forest_snapshot_{chain}_{year}-{month}-{day}_height_{height}.car`
+        /// Date is in ISO 8601 date format.
+        /// Arguments:
+        ///  - chain - chain name e.g. `mainnet`
+        ///  - year
+        ///  - month
+        ///  - day
+        ///  - height - the epoch
+        #[structopt(short, default_value = OUTPUT_PATH_DEFAULT_FORMAT, verbatim_doc_comment)]
+        output_path: PathBuf,
+        /// Skip creating the checksum file.
+        #[structopt(long)]
+        skip_checksum: bool,
+    },
+
+    /// Fetches the most recent snapshot from a trusted, pre-defined location.
+    Fetch {
+        /// Directory to which the snapshot should be downloaded. If not provided, it will be saved
+        /// in default Forest data location.
+        #[structopt(short, long)]
+        snapshot_dir: Option<PathBuf>,
+    },
+
+    /// Shows default snapshot dir
+    Dir,
+
+    /// List local snapshots
+    List {
+        /// Directory to which the snapshots are downloaded. If not provided, it will be the
+        /// default Forest data location.
+        #[structopt(short, long)]
+        snapshot_dir: Option<PathBuf>,
+    },
+
+    /// Remove local snapshot
+    Remove {
+        /// Snapshot filename to remove
+        filename: PathBuf,
+
+        /// Directory to which the snapshots are downloaded. If not provided, it will be the
+        /// default Forest data location.
+        #[structopt(short, long)]
+        snapshot_dir: Option<PathBuf>,
+
+        /// Answer yes to all forest-cli yes/no questions without prompting
+        #[structopt(short, long)]
+        yes: bool,
+    },
+
+    /// Prune local snapshot, keeps the latest only.
+    /// Note that file names that do not match forest_snapshot_{chain}_{year}-{month}-{day}_height_{height}.car
+    /// pattern will be ignored
+    Prune {
+        /// Directory to which the snapshots are downloaded. If not provided, it will be the
+        /// default Forest data location.
+        #[structopt(short, long)]
+        snapshot_dir: Option<PathBuf>,
+
+        /// Answer yes to all forest-cli yes/no questions without prompting
+        #[structopt(short, long)]
+        yes: bool,
```

Comment:
Should we also have `no` to always decline prompts? E.g. you can find a similar approach in e.g. `dnf` (Fedora package manager)
```
❯ dnf install --help | rg assume
                   [--rpmverbosity [debug level name]] [-y] [--assumeno]
  -y, --assumeyes       automatically answer yes for all questions
  --assumeno            automatically answer no for all questions
```

Reply:
I think --no makes sense when there are optional steps, like checking updates, for this case maybe --dry-run is better?

Reply:
Both kind of make sense. We just need to be tactical and not implement stuff that won't be used. If you believe the 99% use case would be to use `-y` then there's no point in spending time on some other flags that won't be used outside of my imagination. :) Up to you! :)

Reply:
Sounds good, then I will not implement `--no` to reduce the code complexity and our maintenance cost, ppl can still use `yes` tool in `coreutils` to achieve this.  `yes n | forest-cli snapshot prune`

Reply:
e.g. podman image prune implements --force only

```
➜  forest git:(subcommand-snapshot) podman image prune -h
Remove unused images

Description:
  Removes dangling or unused images from local storage.

Usage:
  podman image prune [options]

Examples:
  podman image prune

Options:
  -a, --all                  Remove all images not in use by containers, not just dangling ones
      --filter stringArray   Provide filter values (e.g. 'label=<key>=<value>')
  -f, --force                Do not prompt for confirmation

➜  forest git:(subcommand-snapshot) yes n | podman image prune
WARNING! This command removes all dangling images.
Are you sure you want to continue? [y/N] %
➜  forest git:(subcommand-snapshot)
```

Reply:
In general, `yes` could be potentially also used for `--yes` to further reduce code complexity

Reply:
Hmm, just found it's more complicated than I thought, rust stdlib does not provide `stdin().read_char` due to unicode reason, and the console implementation does not work with `yes` tool. I might have to switch back to stdlib to make `yes` tool work.

Reply:
Then maybe let's go with `--force` path. Whatever we choose let's have it uniform across the entire implementation

Reply:
Fixed. I have switch back to using stdlib, and it now works with `yes` tool